### PR TITLE
[NO GBP] Fixes lunatics not having an objective and occasionally not getting their amulettes

### PR DESCRIPTION
**Pull request by Rex9001**
## About The Pull Request
Lunatics now have an objective and it properly displays. 
Lunatics now get a moonlight amulette on the ground if their hand are full.
## Why It's Good For The Game
Lunatics had no idea who their master was except for their flavor text. This made it kinda confusing. This was worsened when they didnt have the identifying amulette.
## Changelog
:cl:
fix: Lunatics spawned from moon ascension now actually have an objective to assist their ringleader
fix: Lunatics now get a moonlight amulette on the ground if they have full hands
/:cl:

### CODE REVIEW

**Comment by necromanceranne** [comment]
If it ends up dropping on the ground, players may still end up leaving it behind. A good example of this is cult daggers on conversion and my original implementation of nukies spawning with tools in a box that dumped everything on the ground. Both instances, players just completely overlook it.

My best suggestion is to therefore either force the amulette into the neck slot, or to force unequip whatever is in someones hands, and then equip the amulette.

**Comment by RikuLeinonenVuoksi** [comment]
> If it ends up dropping on the ground, players may still end up leaving it behind. A good example of this is cult daggers on conversion and my original implementation of nukies spawning with tools in a box that dumped everything on the ground. Both instances, players just completely overlook it.
> 
> My best suggestion is to therefore either force the amulette into the neck slot, or to force unequip whatever is in someones hands, and then equip the amulette.

And if they have, say, the funny necklace that instakills you if you take it off?
Or any other future edgecase where the action of taking it off causes something bad to happen?
Yeah, that's a bit of an issue...

Anywho, just spawn it in their hands if they don't have their neck slot open and if their hands are full, only then spawn it on the ground. That should be enough safeguards to where 99% of people will have it on.

Also, it badly needs a HUD icon since constantly examining people when 70 dudes are scurrying around is just impossible. You could make the HUD icon appear based on whether you have the necklace on or not.

As for any problematic neck items, you COULD make a blacklist and just add a component to the existing neck item to make it glow or smth and drop the old item if it's not in the blacklist. (the blacklist would be quite small, possibly just that 1 necklace)

**Comment by necromanceranne** [comment]
> And if they have, say, the funny necklace that instakills you if you take it off? Or any other future edgecase where the action of taking it off causes something bad to happen? Yeah, that's a bit of an issue...

I think that affects maybe a very extreme minority in a given round. I also don't really foresee much anything akin to the memento mori being added again, so you can just check for that specifically if that is a problem. Undermining design due to very specific edgecases is a bit silly.

**Comment by Rex9001** [comment]
I cannot add hud icons for the time being. There is a feature freeze and that would very much fall under feature, for the same reason I cannot rebalance any parts of this path.

**Comment by Rex9001** [comment]
As for momento mori, I think its a risk the player is gonna have to take, adding an additional blacklist check for one item seems a bit silly to me. Especially when its mostly worn by miners and only people on the station Z can become lunatics

File: code/modules/antagonists/heretic/knowledge/moon_lore.dm [hunk 1]
```diff
@@ -236,8 +236,11 @@
 			continue
 		var/datum/antagonist/lunatic/lunatic = crewmate.mind.add_antag_datum(/datum/antagonist/lunatic)
 		lunatic.set_master(user.mind, user)
-		var/obj/item/clothing/neck/heretic_focus/moon_amulette/moon_amulette = new
-		crewmate.put_in_active_hand(moon_amulette)
+		if(crewmate.get_active_held_item())
+			new /obj/item/clothing/neck/heretic_focus/moon_amulette(crewmate_turf)
+		else
+			var/obj/item/clothing/neck/heretic_focus/moon_amulette/moon_amulette = new
+			crewmate.put_in_active_hand(moon_amulette)
```
Comment (Ghommie) [inline code/modules/antagonists/heretic/knowledge/moon_lore.dm:243]:
```suggestion
		var/obj/item/clothing/neck/heretic_focus/moon_amulette/amulet = new(crewmate_turf)
		var/static/list/slots = list(
			"neck" = ITEM_SLOT_NECK,
			"hands" = ITEM_SLOT_HANDS,
			"backpack" = ITEM_SLOT_BACKPACK,
			"right pocket" = ITEM_SLOT_RPOCKET,
			"left pocket" = ITEM_SLOT_RPOCKET,
		)
		crewmate.equip_in_one_of_slots(amulet, slots, qdel_on_fail = FALSE)
```

Comment (Rex9001) [reply code/modules/antagonists/heretic/knowledge/moon_lore.dm]: Great review, thanks!

**Pull request merged**